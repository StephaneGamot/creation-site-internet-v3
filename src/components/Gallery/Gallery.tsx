"use client";

import Image from "next/image";
import IA from "./../../../public/images/IA.webp";
import IB from "./../../../public/images/IB.webp";
import IC from "./../../../public/images/IC.webp";

export default function Gallery() {
  return (
    <section className="py-10 my-8" >
      <div className="mx-auto max-w-6xl px-6 grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6">
        <div className="overflow-hidden rounded-2xl ">
          <Image
            src={IA}
            alt="Site web créé à propos de l’IA"
            className="w-full h-auto object-cover"
            loading="lazy"
            width={400}
            height={235}
          />
        </div>
        <div className="overflow-hidden rounded-2xl">
          <Image
            src={IB}
            alt="Site e-commerce de bien-être"
            className="w-full h-auto object-cover"
            loading="lazy"
            width={400}
            height={235}
          />
        </div>
        <div className="overflow-hidden rounded-2xl">
          <Image
            src={IC}
            alt="Site internet de recettes culinaires"
            className="w-full h-auto object-cover"
            loading="lazy"
            width={400}
            height={235}
          />
        </div>
      </div>
    </section>
  );
}
