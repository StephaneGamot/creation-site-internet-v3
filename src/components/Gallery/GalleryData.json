[
  {
    "id": 1,
    "src": "./../../../public/images/IA.webp",
    "alt": "Site web créé a props de lIA"
  },
  {
    "id": 2,
 "src": "./../../../public/images/IB.webp",
    "alt": "Site E-commerce de Bien-être"
  },
  {
    "id": 3,
 "src": "./../../../public/images/IC.webp",
    "alt": "Site internet de recettes culinaire"
  }
]