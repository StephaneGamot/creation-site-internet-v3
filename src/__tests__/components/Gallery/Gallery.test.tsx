import React from "react";
import { render, screen } from "@testing-library/react";
import Gallery from "@/components/Gallery/Gallery";
import { axe } from "jest-axe";

// ✅ Mock des images statiques
jest.mock("next/image", () => (props: any) => {
  // On force le rendu des balises img classiques
  return <img {...props} />;
});

jest.mock("../../../public/images/IA.webp", () => "/mock/IA.webp");
jest.mock("../../../public/images/IB.webp", () => "/mock/IB.webp");
jest.mock("../../../public/images/IC.webp", () => "/mock/IC.webp");

describe("Gallery", () => {
  it("affiche les trois images avec les bons alt", () => {
    render(<Gallery />);

    expect(screen.getByAltText("Site web créé à propos de l’IA")).toBeInTheDocument();
    expect(screen.getByAltText("Site e-commerce de bien-être")).toBeInTheDocument();
    expect(screen.getByAltText("Site internet de recettes culinaires")).toBeInTheDocument();
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<Gallery />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const { asFragment } = render(<Gallery />);
    expect(asFragment()).toMatchSnapshot();
  });
});
