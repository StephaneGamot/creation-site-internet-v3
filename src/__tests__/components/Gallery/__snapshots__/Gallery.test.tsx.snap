// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Gallery correspond au snapshot 1`] = `
<DocumentFragment>
  <section
    class="py-10 my-8"
  >
    <div
      class="mx-auto max-w-6xl px-6 grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6"
    >
      <div
        class="overflow-hidden rounded-2xl "
      >
        <img
          alt="Site web créé à propos de l’IA"
          class="w-full h-auto object-cover"
          height="235"
          loading="lazy"
          src="/mock/IC.webp"
          width="400"
        />
      </div>
      <div
        class="overflow-hidden rounded-2xl"
      >
        <img
          alt="Site e-commerce de bien-être"
          class="w-full h-auto object-cover"
          height="235"
          loading="lazy"
          src="/mock/IC.webp"
          width="400"
        />
      </div>
      <div
        class="overflow-hidden rounded-2xl"
      >
        <img
          alt="Site internet de recettes culinaires"
          class="w-full h-auto object-cover"
          height="235"
          loading="lazy"
          src="/mock/IC.webp"
          width="400"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
